src/interpolate.cxx

```diff
@@ -311,6 +311,96 @@ double PiecewiseLogLinearInterpolate::derivative(double x) const
   }
 }
 
+PiecewiseSemiLogLinearInterpolate::PiecewiseSemiLogLinearInterpolate(
```

Comment:
I would change the name to `PiecewiseSemiLogXLinearInterpolate` as `y` is assumed to be in the right dimensions